### PR TITLE
RM-9246 Convert the non-SDK style test sites to SDK-style projects

### PR DESCRIPTION
**Pull request by a-ctor**
https://re-motion.atlassian.net/browse/RM-9246

### CODE REVIEW

File: Remotion/ObjectBinding/Web.Development.WebTesting.TestSite.NetFramework/ObjectBinding.Web.Development.WebTesting.TestSite.NetFramework.csproj [hunk 1]
```diff
@@ -1,231 +1,28 @@
...
-    <Content Include="Web.config" />
+    <ProjectReference Include="..\..\Core\Core\Core.Core.csproj" />
+    <ProjectReference Include="..\..\Development\Web\Development.Web.csproj" />
+    <ProjectReference Include="..\..\Web\Core\Web.Core.csproj" />
+    <ProjectReference Include="..\..\Web\Development.WebTesting.TestSite.Infrastructure\Web.Development.WebTesting.TestSite.Infrastructure.csproj" />
+    <ProjectReference Include="..\Core\ObjectBinding.Core.csproj" />
+    <ProjectReference Include="..\Sample\ObjectBinding.Sample.csproj" />
+    <ProjectReference Include="..\Web.Development.WebTesting.TestSite.Shared\ObjectBinding.Web.Development.WebTesting.TestSite.Shared.csproj" />
+    <ProjectReference Include="..\Web.UnitTests\ObjectBinding.Web.UnitTests.csproj" />
+    <ProjectReference Include="..\Web\ObjectBinding.Web.csproj" />
```
Comment (MichaelKetting) [inline Remotion/ObjectBinding/Web.Development.WebTesting.TestSite.NetFramework/ObjectBinding.Web.Development.WebTesting.TestSite.NetFramework.csproj:23]: I think we could probably drop sevreal of the project refences thanks to transitive refs?

File: Remotion/Web/Development.WebTesting.TestSite.NetFramework/Web.Development.WebTesting.TestSite.NetFramework.csproj [hunk 2]
```diff
@@ -1,195 +1,25 @@
...
-    <ProjectReference Include="..\Development.WebTesting.TestSite.Shared\Web.Development.WebTesting.TestSite.Shared.csproj">
-      <Project>{165a8123-c506-4239-bb21-adbef603bc0b}</Project>
-      <Name>Web.Development.WebTesting.TestSite.Shared</Name>
-    </ProjectReference>
+    <ProjectReference Include="..\..\Core\Core\Core.Core.csproj" />
+    <ProjectReference Include="..\..\Development\Web\Development.Web.csproj" />
+    <ProjectReference Include="..\..\Security\Core\Security.Core.csproj" />
+    <ProjectReference Include="..\Core\Web.Core.csproj" />
+    <ProjectReference Include="..\Development.WebTesting.TestSite.Infrastructure\Web.Development.WebTesting.TestSite.Infrastructure.csproj" />
+    <ProjectReference Include="..\Development.WebTesting.TestSite.Shared\Web.Development.WebTesting.TestSite.Shared.csproj" />
```
Comment (MichaelKetting) [inline Remotion/Web/Development.WebTesting.TestSite.NetFramework/Web.Development.WebTesting.TestSite.NetFramework.csproj:20]: Same question as before: could/should we utilize tansitive deps?

File: Remotion/Web/Test.NetFramework/Web.Test.NetFramework.csproj [hunk 3]
```diff
@@ -1,220 +1,26 @@
...
-  </ItemGroup>
-  <ItemGroup>
-    <Service Include="{B4F97281-0DBD-4835-9ED8-7DFB966E87FF}" />
+    <ProjectReference Include="..\..\Core\Core\Core.Core.csproj" />
+    <ProjectReference Include="..\..\Data\Core\Data.Core.csproj" />
+    <ProjectReference Include="..\..\Development\Web\Development.Web.csproj" />
+    <ProjectReference Include="..\..\Globalization\Core\Globalization.Core.csproj" />
+    <ProjectReference Include="..\..\Security\Core\Security.Core.csproj" />
+    <ProjectReference Include="..\Core\Web.Core.csproj" />
+    <ProjectReference Include="..\Test.Shared\Web.Test.Shared.csproj" />
```
Comment (MichaelKetting) [inline Remotion/Web/Test.NetFramework/Web.Test.NetFramework.csproj:21]: transitive refs: just for completeness' sake :)

File: SecurityManager/Clients.Web.Test/SecurityManager.Clients.Web.Test.csproj [hunk 4]
```diff
@@ -1,234 +1,38 @@
...
+    <ProjectReference Include="..\..\Remotion\Mixins\Core\Mixins.Core.csproj" />
+    <ProjectReference Include="..\..\Remotion\ObjectBinding\Core\ObjectBinding.Core.csproj" />
+    <ProjectReference Include="..\..\Remotion\ObjectBinding\Validation\ObjectBinding.Validation.csproj" />
+    <ProjectReference Include="..\..\Remotion\ObjectBinding\Web.Validation\ObjectBinding.Web.Validation.csproj" />
+    <ProjectReference Include="..\..\Remotion\ObjectBinding\Web\ObjectBinding.Web.csproj" />
+    <ProjectReference Include="..\..\Remotion\Security\Core\Security.Core.csproj" />
+    <ProjectReference Include="..\..\Remotion\Web\Core\Web.Core.csproj" />
+    <ProjectReference Include="..\Clients.Web\SecurityManager.Clients.Web.csproj" />
+    <ProjectReference Include="..\Core\SecurityManager.Core.csproj" />
+  </ItemGroup>
```
Comment (MichaelKetting) [inline SecurityManager/Clients.Web.Test/SecurityManager.Clients.Web.Test.csproj:34]: transitive refs: just for completeness' sake :)

**Review by MichaelKetting — APPROVED** [review]

File: Remotion/ObjectBinding/Web.Development.WebTesting.TestSite.NetFramework/ObjectBinding.Web.Development.WebTesting.TestSite.NetFramework.csproj (hunk 1 above)
Comment (a-ctor) [reply Remotion/ObjectBinding/Web.Development.WebTesting.TestSite.NetFramework/ObjectBinding.Web.Development.WebTesting.TestSite.NetFramework.csproj:23]: Transitive references don't seem to work with .NET Framework so we still have to reference most of them. While there are a few "unused" references per project I would keep them in as well as somebody might depend on them at runtime. On develop, we convert these projects to .NET soon anyway. So I would leave them as is for compatibility in the backports.

**Pull request merged**